### PR TITLE
Richard/delete, update and edit using observables

### PR DESCRIPTION
**Pull request by kokobear3327**

### CODE REVIEW

File: src/app/admin/product-form/product-form.component.html [hunk 1]
```diff
@@ -89,14 +89,23 @@
         </div>
       </div>
       <button class="btn-primary">Save</button>
+      <br />
```
Comment (ahaly92) [inline src/app/admin/product-form/product-form.component.html]: use styling over multiple <br/> tags

File: src/app/admin/product-form/product-form.component.ts [hunk 2]
```diff
@@ -2,29 +2,49 @@ import { CategoryService } from './../../category.service';
...
+    private route: ActivatedRoute,
     private productService: ProductService,
     private categoryService: CategoryService
   ) {
     this.categoriesObservable = categoryService.getCategories();
+    this.id = this.route.snapshot.paramMap.get('id');
+    if (this.id)
+      this.subscription = this.productService
+        .get(this.id)
+        .subscribe(product => (this.product = product));
```
Comment (ahaly92) [inline src/app/admin/product-form/product-form.component.ts:31]: this is simple enough for async pipe no?

File: src/app/admin/product-form/product-form.component.ts [hunk 3]
```diff
@@ -2,29 +2,49 @@ import { CategoryService } from './../../category.service';
...
   }
+
+  delete() {
+    if (confirm('Are you sure you want to delete this product? ')) {
+      this.productService.delete(this.id);
+      this.router.navigate(['admin/products/manage']);
+    }
+  }
+
+  ngOnDestroy() {
```
Comment (ahaly92) [inline src/app/admin/product-form/product-form.component.ts:47]: if we go the async route lose the ngOnDestroy

File: src/app/models/product.ts [hunk 4]
```diff
@@ -1,5 +1,5 @@
 export interface Product {
-  key: string;
+  $key: string;
```
Comment (ahaly92) [inline src/app/models/product.ts]: key is more consistent with the the rest of the code. lose $

File: src/app/product.service.ts [hunk 5]
```diff
@@ -19,4 +20,33 @@ export class ProductService {
...
+
+  getProductsRaw() {
+    return this.database
+      .list('/products')
+      .snapshotChanges()
+      .pipe(
+        map(changes => {
+          return changes.map(rawProduct => {
+            const data = rawProduct.payload.val();
+            const $key = rawProduct.payload.key;
```
Comment (ahaly92) [inline src/app/product.service.ts]: change to key

File: src/app/admin/product-form/product-form.component.ts (hunk 2 above)
Comment (kokobear3327) [reply src/app/admin/product-form/product-form.component.ts:31]: There is not place in the html to put the async pipe?  This is drawing the id from the product.

File: src/app/admin/admin-products/admin-products.component.ts [hunk 6]
```diff
@@ -1,7 +1,7 @@
 import { Product } from './../../models/product';
 import { Observable } from 'rxjs';
 import { ProductService } from './../../product.service';
-import { Component } from '@angular/core';
+import { Component, OnDestroy } from '@angular/core';
```
Comment (ahaly92) [inline src/app/admin/admin-products/admin-products.component.ts]: are you using this OnDestroy anymore? If not clean up imports

File: src/app/admin/product-form/product-form.component.ts (hunk 2 above)
Comment (ahaly92) [reply src/app/admin/product-form/product-form.component.ts:31]: 💯

**Comment by ahaly92** [comment]
one small change but then :shipit:

**Pull request merged**